src/endpoint/s3/ops/s3_post_object_restore.js

```diff
@@ -1,32 +1,44 @@
 /* Copyright (C) 2023 NooBaa */
 'use strict';
 
+const { S3Error } = require('../s3_errors');
 const s3_utils = require('../s3_utils');
 
 /**
  * https://docs.aws.amazon.com/AmazonS3/latest/API/API_RestoreObject.html
  */
 async function post_object_restore(req, res) {
     const encryption = s3_utils.parse_encryption(req);
-    const days = req.body?.RestoreRequest?.Days ? parseInt(req.body.RestoreRequest.Days?.[0], 10) : undefined;
+
+    if (!req.body?.RestoreRequest?.Days?.[0]) {
+        throw new S3Error(S3Error.MalformedXML);
+    }
+
+    const days = parseInt(req.body.RestoreRequest.Days[0], 10);
```

Comment:
notice these differences - https://medium.com/@cristi.nord/js-string-to-number-7d82010ef263
parseInt('123abc', 10) returns 123
Number('123abc') returns NaN
Number('0x123') returns 291
Maybe we just want s3_utils.parse_decimal_int() to hide these implementation details and then use parseInt() inside + check that the string contains only digits...

Reply:
Added a function in the `s3_utils`